contrib/initramfs/zfsunlock

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -eu
+echo "Waiting for zfs init script to complete, press Ctrl-C to exit into shell ..."
+while [ ! -e /run/zfs_fs_name ] && [ ! -e /run/zfs_unlock_complete ]; do
+	sleep 1;
+done
+if [ -e /run/zfs_unlock_complete ]; then
+	exit 0
+fi
+echo
+echo "Unlocking encrypted ZFS pools ..."
+echo "Enter the password or press Ctrl-C to exit into shell."
+echo
+zfs_fs_name=""
+fs_password_accepted=false
+while true; do
+	while [ "$fs_password_accepted" = "true" ] && [ "$(cat /run/zfs_fs_name)" = "$zfs_fs_name" ] && [ ! -e /run/zfs_unlock_complete ]; do
+		sleep 1;
+        done
+	if [ -e /run/zfs_unlock_complete ]; then
+		break
+	fi
+	zfs_fs_name=$(cat /run/zfs_fs_name)
+	zfs_console_askpwd_cmd=$(cat /run/zfs_console_askpwd_cmd)
+	systemd-ask-password "Encrypted ZFS password for ${zfs_fs_name}:" | \
+		/sbin/zfs load-key "$zfs_fs_name"
```

Comment:
I haven't tested `zfs load-key`'s exit status. If you enter the wrong password, does it exit non-zero? If so, the new `set -e` is going to break this in the case of an incorrect password.

Reply:
yep, it does, there are 3 diffirent exit codes, one for incorrect key, one for master key unavailable and one for uncaught exception